src/Microsoft.CodeAnalysis.Analyzers/UnitTests/MetaAnalyzers/CompareSymbolsCorrectlyTests.cs

```diff
@@ -228,6 +306,35 @@ class TestClass {{
 ";
             var fixedSource = $@"
 using Microsoft.CodeAnalysis;
+class TestClass {{
+    bool Method(Symbol x, {symbolType} y) {{
+        return SymbolEqualityComparer.Default.Equals(x, y);
+    }}
+}}
+";
+
+            await new VerifyCS.Test
+            {
+                TestState = { Sources = { source, MinimalSymbolImplementationCSharp, SymbolEqualityComparerStubCSharp } },
+                FixedState = { Sources = { fixedSource, MinimalSymbolImplementationCSharp, SymbolEqualityComparerStubCSharp } },
+            }.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(nameof(ISymbol))]
+        [InlineData(nameof(INamedTypeSymbol))]
+        public async Task CompareSymbolImplementationWithInterface_NoComparer_CSharp(string symbolType)
+        {
+            var source = $@"
+using Microsoft.CodeAnalysis;
+class TestClass {{
+    bool Method(Symbol x, {symbolType} y) {{
+        return [|x == y|];
```

Comment:
Do we have a unit test where input source has `Equals(x, y);`, symbol equality comparer is available and we offer a diagnostic/fix to change it to `SymbolEqualityComparer.Default.Equals(x, y);`?